simul/platform/aws/sshController.go

```diff
@@ -85,24 +84,40 @@ func copyFile(sftpClient *sftp.Client, file string) error {
 }
 
 //Run runs command on a remote host using ssh and waits for output
-func (sshCMD *sshController) Run(command string) (string, error) {
-	//fmt.Println(">>>> Runnning >>>> ", command)
-	var stdoutBuf bytes.Buffer
+func (sshCMD *sshController) Run(command string) (io.Reader, error) {
 	session, err := sshCMD.client.NewSession()
-
-	session.Stdout = &stdoutBuf
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
+	// +++ We need to be careful here:
+	// From the doc:
+	// If the StdoutPipe reader is
+	// not serviced fast enough it may eventually cause the
+	// remote command to block.
+	outPipe, err := session.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+	errPipe, err := session.StderrPipe()
+	if err != nil {
+		return nil, err
+	}
+	pipe := io.MultiReader(outPipe, errPipe)
+	if err != nil {
+		return nil, err
+	}
+	// +++Seems like even after closing ssh-session we can recive data (it works)
+	// From the doc:
+	// Close signals end of channel use. No data may be sent after this call.
```

Comment:
So there are 2 things, ssh session (one per ssh command) and long-lived ssh connection.
Ssh session is closed after every command and opens for every new command. I guess we can recive data because the ssh-connection is still open (I close it only after all the ssh communication with EC2 is done).